packages/hash/frontend/src/pages/[account-slug]/[page-slug].page.tsx

```diff
@@ -39,50 +50,143 @@ import { PageIconButton } from "../../components/PageIconButton";
+  typeof queryParams["page-slug"] === "string";
+
+export const parsePageUrlQueryParams = (params: PageParsedUrlQuery) => {
+  const workspaceShortname = params["account-slug"].slice(1);
+
+  const pageEntityUuid = params["page-slug"];
+
+  return { workspaceShortname, pageEntityUuid };
+};
+
 /**
  * This is used to fetch the metadata associated with blocks that're preloaded
  * ahead of time so that the client doesn't need to
  *
  * @todo Include blocks present in the document in this
  */
-export const getStaticProps: GetStaticProps<PageProps> = async () => {
+export const getServerSideProps: GetServerSideProps<PageProps> = async ({
+  req,
+  params,
+}) => {
   const fetchedBlocks = await Promise.all(
     defaultBlockComponentIds.map((componentId) => fetchBlock(componentId)),
   );
 
+  if (!params || !tbdIsPageParsedUrlQuery(params)) {
+    throw new Error(
+      "Invalid page URL query params passed to `getServerSideProps`.",
+    );
+  }
+
+  const { workspaceShortname, pageEntityUuid } =
+    parsePageUrlQueryParams(params);
+
+  const cookieString = req?.headers.cookie;
+
+  const workspacesSubgraph = await apolloClient
+    .query<GetAllLatestEntitiesQuery>({
+      query: getAllLatestEntitiesQuery,
+      variables: {
+        rootEntityTypeIds: [
+          types.entityType.user.entityTypeId,
+          types.entityType.org.entityTypeId,
+        ],
+        constrainsValuesOn: { outgoing: 0 },
+        constrainsPropertiesOn: { outgoing: 0 },
+        constrainsLinksOn: { outgoing: 0 },
+        constrainsLinkDestinationsOn: { outgoing: 0 },
+        isOfType: { outgoing: 1 },
+        hasLeftEntity: { incoming: 1, outgoing: 1 },
+        hasRightEntity: { incoming: 1, outgoing: 1 },
+      },
+      context: { headers: { cookie: cookieString } },
+    })
+    .then(({ data }) => data.getAllLatestEntities);
+
+  const workspaces = getRootsAsEntities(workspacesSubgraph).map((entity) =>
+    entity.metadata.entityTypeId === types.entityType.user.entityTypeId
+      ? constructMinimalUser({
+          userEntityEditionId: entity.metadata.editionId,
+          subgraph: workspacesSubgraph,
+        })
+      : constructMinimalOrg({
+          orgEntityEditionId: entity.metadata.editionId,
+          subgraph: workspacesSubgraph,
+        }),
+  );
+
+  /**
+   * @todo: filtering all workspaces by their shortname should not be happening
+   * client side. This could be addressed by exposing structural querying
+   * to the frontend.
+   *
+   * @see https://app.asana.com/0/1201095311341924/1202863271046362/f
+   */
+  const pageWorkspace = workspaces.find(
+    (workspace) => workspace.shortname === workspaceShortname,
+  );
+
+  if (!pageWorkspace) {
+    throw new Error(
+      `Could not find page workspace with shortname "${workspaceShortname}".`,
+    );
+  }
+
+  const pageOwnedById =
+    pageWorkspace.kind === "user"
+      ? pageWorkspace.userAccountId
+      : pageWorkspace.orgAccountId;
```

Comment:
Changing the property name to `accountId` would simplify this as well

Reply:
Addressed in https://github.com/hashintel/hash/pull/1650/commits/936a4734986f4d4cdfdc74feb321c997728a2e8c